File: app/src/main/java/ru/eugen/noteseugen/data/Card.java
package ru.eugen.noteseugen.data;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.Date;

public class Card implements Parcelable {
    private String note;
    private String dateS;
    private String essence;
    private Date date;


    public Card(String note, String essence, Date date) {
        this.note = note;
        this.essence = essence;
        this.date = date;
    }

    protected Card(Parcel in) {
        note = in.readString();
        essence = in.readString();
        date = new Date(in.readLong());
    }

    public static final Creator<Card> CREATOR = new Creator<Card>() {
        @Override
        public Card createFromParcel(Parcel in) {
            return new Card(in);
        }

        @Override
        public Card[] newArray(int size) {
            return new Card[size];
        }
    };

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public void setDate(String date) {
        this.dateS = date;
    }

    public String getEssence() {
        return essence;
    }

    public void setEssence(String essence) {
        this.essence = essence;
    }
    public Date getDate() {
        return date;
    }
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(note);
        dest.writeString(dateS);
        dest.writeString(essence);
        dest.writeLong(date.getTime());
    }
}

File: app/src/main/java/ru/eugen/noteseugen/data/Card.kt
package ru.eugen.noteseugen.data

import android.os.Parcel
import android.os.Parcelable

import java.util.Date

class Card : Parcelable {
    var id:String
    var note:String
    var essence:String
    var date:Date

    constructor(note:String, essence:String, date:Date ) {
        this.note = note
        this.essence = essence
        this.date = date
        this.id = ""
    }

    constructor (in2:Parcel) {
        id = in2.readString()?:""
        note = in2.readString()?:""
        essence = in2.readString()?:""
        date = Date(in2.readLong())
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeString(note)
        dest.writeString(essence)
        dest.writeLong(date.getTime())
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Card> {
        override fun createFromParcel(in2: Parcel): Card {
            return Card(in2)
        }

        override fun newArray(size: Int): Array<Card?> {
            return arrayOfNulls(size)
        }
    }
}
